src/range_statement.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 The Tari Project
+// SPDX-License-Identifier: BSD-3-Clause
+
+#![deny(missing_docs)]
+
+//! Bulletproof+ generators, vector of commitments, vector of optional minimum promised
+//! values and a vector of optional seed nonces for mask recovery
+
+use curve25519_dalek::{ristretto::RistrettoPoint, scalar::Scalar};
+use zeroize::Zeroize;
+
+use crate::{errors::ProofError, range_parameters::RangeParameters};
+
+/// The range proof statement contains the generators, vector of commitments, vector of optional minimum promised
+/// values and a vector of optional seed nonces for mask recovery
+#[derive(Clone, Debug)]
+pub struct RangeStatement {
+    /// The generators and base points needed for a batch of range proofs
+    pub generators: RangeParameters,
+    /// The batch of commitments
+    pub commitments: Vec<RistrettoPoint>,
+    /// Optional minimum promised values
+    pub minimum_value_promises: Vec<Option<u64>>,
+    /// Optional seed nonces for mask recovery
+    pub seed_nonce: Option<Scalar>,
+}
+
+impl RangeStatement {
+    /// Initialize a new 'RangeStatement' with sanity checks
+    pub fn init(
+        generators: RangeParameters,
+        commitments: Vec<RistrettoPoint>,
+        minimum_value_promise: Vec<Option<u64>>,
+        seed_nonce: Option<Scalar>,
+    ) -> Result<RangeStatement, ProofError> {
+        if !commitments.len().is_power_of_two() {
+            return Err(ProofError::InternalDataInconsistent(
```

Comment:
This is used a lot, perhaps a better name would be `ProofError::InvalidArgument`